components/input-field/InputField.tsx

```diff
@@ -21,16 +22,16 @@ const InputField = ({
   ): void => {
     if (setValue)
       if (!nesting) {
-        setValue((oldValues: UserInput) => ({
+        setValue((oldValues) => ({
           ...oldValues,
-          [id as string]: event.target.value,
+          [id]: event.target.value,
         }));
       } else {
-        setValue((oldValues: UserInput) => ({
+        setValue((oldValues) => ({
           ...oldValues,
-          [nesting as string]: {
-            ...(oldValues[nesting] as FullName),
-            [id as string]: event.target.value,
+          [nesting]: {
+            ...(oldValues[nesting] as Record<string, unknown>),
```

Comment:
Record<string, unknown> means that this is an object with strings as keys that have unknown values, right? I am just wondering whether there is an easier or more explicit way to do this. But also fine if we keep it like that.

Reply:
since this is supposed to be a very generally usable component I changed the type to the most general (= least specific) type that still works here. Record<string, unknown> is the suggested workaround for simply using object here (because according to eslint there are some issues with this), which seems appropriate because all we need to know about oldValues[nesting] is an object (and, hence, spreadable).

Reply:
okay, that's fine. Thanks